pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -472,7 +472,10 @@ async def _map_messages(self, messages: list[ModelMessage]) -> tuple[ContentDict
                     message_parts = [{'text': ''}]
                 contents.append({'role': 'user', 'parts': message_parts})
             elif isinstance(m, ModelResponse):
-                contents.append(_content_model_response(m))
+                model_content = _content_model_response(m)
+                # Skip model responses with empty parts (e.g., thinking-only responses)
```

Comment:
@ethanabrooks Does this mean this would be a non-issue once we start sending back thinking parts? (https://github.com/pydantic/pydantic-ai/issues/2594) I mean to work on that next week